src/oneD/StFlow.cpp

```diff
@@ -629,11 +638,22 @@ void StFlow::updateTransport(double* x, size_t j0, size_t j1)
         for (size_t j = j0; j < j1; j++) {
             setGasAtMidpoint(x,j);
             m_visc[j] = (m_dovisc ? m_trans->viscosity() : 0.0);
-            m_trans->getMixDiffCoeffs(&m_diff[j*m_nsp]);
+
+            if (m_fluxGradientBasis == ThermoBasis::molar) {
+                m_trans->getMixDiffCoeffs(&m_diff[j*m_nsp]);
+            } else {
+                m_trans->getMixDiffCoeffsMass(&m_diff[j*m_nsp]);
+            }
+
             double rho = m_thermo->density();
             double wtm = m_thermo->meanMolecularWeight();
+
             for (size_t k=0; k < m_nsp; k++) {
-                m_diff[k+j*m_nsp] *= m_wt[k] * rho / wtm;
+                if (m_fluxGradientBasis == ThermoBasis::molar) {
+                    m_diff[k+j*m_nsp] *= m_wt[k] * rho / wtm;
+                } else {
+                    m_diff[k+j*m_nsp] *= rho;
+                }
             }
```

Comment:
At least for this inner loop, I think it's worth inverting the structure to check the `m_fluxGradientBasis` before the loop.